### DISCUSSION TITLE
Changelog Causes Confusion

### DISCUSSION THREAD

**Discussion by Wajinn** [vlang/v#12994] [General]
From the "outside looking in", the changelog gives the impression that the last significant update of vlang was V 0.2.3 around Jan 2021.  That will be closing in at around a year ago.  Clearly there are weekly updates and progress is being made (salute to team V), but it's not clear how that fits into the total picture because the last changelog update on vlang.io was a year ago.  

The lack of updating the changelog appears to create confusion.  A person would expect that vlang is at V 0.2.4 - V 0.2.5 by now, but there is nothing in the documentation to confirm such.  It might be helpful to have some more regular pattern of updating the changelog.  Maybe once a month or at least every 2 months?

**Comment by ylluminate** [comment]
I suppose something may need to be done in this regard... I think we need some thoughts or input though since it's a little overwhelming to maintain something like a changelog when everyone is so busy and then going through the changes can similarly be daunting/grinding work that sometimes feels less important.

So I can't help but wonder if perhaps we might be able to come up with a way to have PR submissions have some kind of a categorization or ranking methodology that can be used to build up points of a changelog more readily insomuch that minor points/changes aggregate under a parent item and so forth.  

Obviously I'm just spitting out thoughts and don't have anything concrete in mind yet and don't have time at the moment to ponder on this further, but I do think it's worth some consideration to see if something like this has been done elsewhere and/or how we can do something effectively without really putting a burden on anyone squarely while simultaneously being able to provide meaningful data for perhaps either autogenerating or manually-with-very-very-little-effort-generating something... Hmmm.

**Reply by ghost** [reply]
There seems to be a variety of tools for automatic generation of changelog.

**Reply by ylluminate** [reply]
True, I guess I have been unimpressed or somewhat disappointed by most autogenerated changelogs I've seen in the past due to verbosity or lack of meaningful presentation / coherence... Perhaps there's a good way around this though that I've just not seen/explored.

Personally I don't believe that suggestions like this are sufficient: https://mokkapps.de/blog/how-to-automatically-generate-a-helpful-changelog-from-your-git-commit-messages/

This seems perhaps better, but still not a silver bullet: https://blog.mozilla.org/webdev/2016/07/15/auto-generating-a-changelog-from-git-history/

Perhaps a *secondary* (additional) "semantic versioning" that classifies feature type or using keywords similar to the Mozilla method might help to rank pertinent message info (metadata+message) so as to push certain things into the changelog.  

The trick would then be to actually get folks to do this without having to beat people over the head constantly.  I don't think we want to trade one headache for the other, so we'd need to solve both issues simultaneously in an amicable way.  

Many (most?) commits probably should not even go into a changelog since many folks (developers) use the git history as a sort of changelog via grep, etc... Thus some people may forget to commit meaningful messages that will get parsed when they make a commit that should be aggregated into the changelog...

**Comment by tzSharing** [comment]
I agree, my suggestions：
- update the change log once a month.
- update the project roadmap once a quarter.
